modules/conversantAnalyticsAdapter.js

```diff
@@ -0,0 +1,564 @@
+      const timeoutKey = cnvrHelper.getLookupKey(auctionId, adUnit.code, bid.bidder);
+      if (cnvrHelper.timeoutCache[timeoutKey]) {
+        cnvrAdUnit.bids[bid.bidder].eventCodes.push(CNVR_CONSTANTS.TIMEOUT);
+        cnvrAdUnit.bids[bid.bidder].timeToRespond = args.timeout; // set to Auction defined timeout amount
+        delete cnvrHelper.timeoutCache[timeoutKey];
+      }
+    });
+
+    // Ad media types for the ad slot
+    if (cnvrHelper.keyExistsAndIsObject(adUnit, 'mediaTypes')) {
+      Object.entries(adUnit.mediaTypes).forEach(([mediaTypeName]) => {
+        cnvrAdUnit.mediaTypes.push(mediaTypeName);
+      });
+    }
+
+    // Ad sizes listed under the size key
+    if (Array.isArray(adUnit.sizes) && adUnit.sizes.length >= 1) {
+      adUnit.sizes.forEach(size => {
+        if (!Array.isArray(size) || size.length !== 2) {
+          utils.logMessage(CNVR_CONSTANTS.LOG_PREFIX + 'Unknown object while retrieving adUnit sizes.', adUnit);
+          return; // skips to next item
+        }
+        cnvrAdUnit.sizes.push(cnvrHelper.createAdSize(size[0], size[1]));
+      });
+    }
+
+    // If the Ad Slot is not unique then ad sizes and media types merge them together
+    if (auctionEndPayload.adUnits[adUnit.code]) {
+      // Merge ad sizes
+      Array.prototype.push.apply(auctionEndPayload.adUnits[adUnit.code].sizes, cnvrAdUnit.sizes);
+      // Merge mediaTypes
+      Array.prototype.push.apply(auctionEndPayload.adUnits[adUnit.code].mediaTypes, cnvrAdUnit.mediaTypes);
+    } else {
+      auctionEndPayload.adUnits[adUnit.code] = cnvrAdUnit;
+    }
+  });
+
+  if (Array.isArray(args.noBids)) {
+    args.noBids.forEach(noBid => {
+      const bidPayload = utils.deepAccess(auctionEndPayload, 'adUnits.' + noBid.adUnitCode + '.bids.' + noBid.bidder);
+
+      if (bidPayload) {
+        bidPayload.eventCodes.push(CNVR_CONSTANTS.NO_BID);
+        bidPayload.timeToRespond = 0; // no info for this, would have to capture event and save it there
+      } else {
+        utils.logMessage(CNVR_CONSTANTS.LOG_PREFIX + 'Unable to locate bid object via adUnitCode/bidderCode in payload for noBid reply in END_AUCTION', Object.assign({}, noBid));
+      }
+    });
+  } else {
+    utils.logError(CNVR_CONSTANTS.LOG_PREFIX + 'onAuctionEnd(): noBids not defined in arguments.');
+  }
+
+  // Get bid data from bids sent
+  if (Array.isArray(args.bidsReceived)) {
+    args.bidsReceived.forEach(bid => {
+      const bidPayload = utils.deepAccess(auctionEndPayload, 'adUnits.' + bid.adUnitCode + '.bids.' + bid.bidderCode);
+      if (bidPayload) {
+        bidPayload.eventCodes.push(CNVR_CONSTANTS.BID);
+        bidPayload.timeToRespond = bid.timeToRespond;
+        bidPayload.originalCpm = bid.originalCpm;
+        bidPayload.cpm = bid.cpm;
+        bidPayload.currency = bid.currency;
+        bidPayload.mediaType = bid.mediaType;
+        bidPayload.adSize = {
+          'w': bid.width,
+          'h': bid.height
+        };
```

Comment:
Like here for example,

if for an adUnit you had 3 bids from bidder IX,

The last bid in the array would set it's alues to the bidPayloads object.

Reply:
I think at least, I may be missing something!

Reply:
@robertrmartinez interesting edge case that we didn't realize.  I'm discussing with my team how to handle this and will follow up in the next day or so.  Thanks for your detailed look at our adapter!